csrc/host_ir/executor.h

```diff
@@ -87,7 +87,8 @@ class HostIrExecutor final : public OptInDispatch {
   // Cache Fusions, FusionExecutors
   std::unordered_map<HostUnit*, FusionExecutor> fe_;
   std::unordered_map<HostUnit*, FusionExecutorCache> fec_;
-  std::unordered_map<Stream*, c10::cuda::CUDAStream> streams_;
+  std::unordered_map<Stream*, c10::cuda::CUDAStream> streams_by_pointer_;
+  std::unordered_map<int64_t, c10::cuda::CUDAStream> streams_by_index_;
```

Comment:
Sorry if https://github.com/NVIDIA/Fuser/pull/2621#discussion_r1692592230 wasn't clear. I meant for it to be a `std::unordered_map<int64_t, Stream*>`. This way, we'll still be able to locate `Stream*`s by indices.

Reply:
Sorry, can you clarify what you suggest? I am not sure to understand.

Reply:
I now understood in your implementation there can be multiple `Stream*`s mapping to the same index, so a map from index to a `Stream*` doesn't make sense here. I ran into a different mental model where `Stream*`s are cached, and therefore confused myself.

Reply:
That also justifies the use of std::variant, so reverting 3f84fa6fafa8d074931dbe99f6e819772bff462b sounds good!

Reply:
ok, sounds good